COMMUNICATIONS.md

```diff
@@ -0,0 +1,94 @@
+# OpenSearch Project Communication 
+
+- [Overview](#overview)
+- [Public Slack](#public-slack)
+  - [Getting Started](#getting-started)
+  - [Workspace Channels](#workspace-channels)
+  - [Tips](#tips)
+
+## Overview
+
+The purpose of this document is to provide references and information regarding
+the communication channels for the OpenSearch Project. All communication is
+subject to the [OpenSearch Code of Conduct](CODE_OF_CONDUCT.md). Please see
+[CONTRIBUTING](CONTRIBUTING.md) if you're interested in contributing to the
+project.
+
+## Public Slack
+
+OpenSearch has a workspace on [Slack](https://opensearch.slack.com) to provide
+public communication channels for all people interested in OpenSearch.
+
+The following guidelines include a Getting Started for steps to register and
+setup the workspace along with tips for using the workspace. Please read through
+these guidelines carefully and thoroughly. All communications and behavior on
+the public slack workspace is subject to the 
+[OpenSearch Code of Conduct](CODE_OF_CONDUCT.md).
+
+### Getting Started
+
+_Join Us_
+
+  * Register your account at https://opensearch.slack.com. You will be required
+    to enroll in two-factor authentication to ensure the best possible security
+    of your account.
+
+_Update Profile_
+
+  * Add your interests or keywords in the “What I do” section of your profile 
+    along with your Title and Company/Organization. This is how participants 
+    can network and connect with each other by common interests. No anonymous 
+    display names are allowed.
+  * Feel free to update your profile photo to something unique to you. All 
+    profile photos are subject to the 
+    [OpenSearch Code of Conduct](CODE_OF_CONDUCT.md) policies.
+  * Optionally share your pronouns by adding them to the end of your Full Name 
+    in your profile. This way other participants will be able to see them and 
+    be respectful to your communication preferences. Please also be respectful 
+    of other participants’ pronouns.
+
+### Workspace Channels
+
+After you join OpenSearch workspace you will be automatically added to the 
+following channels:
+
+  * **#general** - a channel for asking general questions. Based on the question 
+    or discussion users will mostly likely be directed to more specific 
+    channels based on the topics of interest.
+  * **#random** - consider this the water cooler in a face to face workspace. 
+    Feel free to introduce yourself to the community, share what you’re 
+    working on, or what you’re excited about learning with OpenSearch.
+
+There are dedicated channels for the following (note that participants might 
```

Comment:
I'd encourage you to start with just #random and #general. Bifurcating where the community interacts prematurely impedes progress IMO. Wait until things are so busy that you need, and then slowly add channels as existing channels grow too busy and people complain.

Reply:
We have pretty large activity on several [OpenSearch repositories](https://github.com/opensearch-project/) including a lot of [dashboard communication](https://github.com/opensearch-project/OpenSearch-Dashboards/issues) which is abstracting the back-end from OpenSearch to accommodate other data providers. This was the idea behind separating channels for OpenSearch and Dashboards but not getting so crazy that we have a channel for every repository. On day 0 we're most certainly going to have a mix mash of discussions on #general with the existing opensearch and dashboard maintainers that I'm afraid will lead to back channeling to avoid noise. 

Would a good compromise be to start with something smaller like #general, #core, #dashboards, #build, and a private #maintainers channel to discuss candidate maintainers?

Reply:
+1 to you both. Hesitate to plan your community, let it be organic, however if your community has already been organic in A.N.Other location, then mimicking the core of that makes sense.

Reply:
From a community standpoint, when starting out I would recommend the approach that less is more. Before I came over to the project - people would add a new category for every different plugin, etc - we now have 30+ categories on the forum, and people typically use about 5 of them. (and now I have a massive cleanup project to make it more functional)

Reply:
What about an #announcements channel that is expected to be low traffic, for major announcements such as new releases or significant deprecations or other important milestones? Given Slack's limitation of only one write-limited channel, this is one channel that needs to be set up out of the gate as an autosubscribe channel so as to avoid the need to ask people to subscribe to it in the future.